src/items.rs

```diff
@@ -225,10 +227,7 @@ impl<'a> FmtVisitor<'a> {
         }
 
         // A conservative estimation, to goal is to be over all parens in generics
-        let args_start = generics.ty_params
-                                 .last()
-                                 .map(|tp| end_typaram(tp))
-                                 .unwrap_or(span.lo);
+        let args_start = generics.ty_params.last().map(|tp| end_typaram(tp)).unwrap_or(span.lo);
```

Comment:
In this case I don't think the motivation of working on garbage is very helpful - in the case that we start with unformatted text (i.e., no whitespace) we would do exactly what we do without the heuristic. In the case that there is random formatting, we would still reformat to make it look nice but we might end up with a multi-line chain rather than a one-line chain.

So, I guess this should be an option, but I would like the default to be to preserve.

I'd be keen to see this done as part of this PR because once we run this without the check, then we lose that info forever and subsequent reformatting cannot put it back